components/base/sensorcontrolled/sensorcontrolled.go

```diff
@@ -208,6 +210,10 @@ func (sb *sensorBase) Reconfigure(ctx context.Context, deps resource.Dependencie
 		}
 		// relock the mutex after setting up the control loop since there is still a  defer unlock
 		sb.mu.Lock()
+
+		go func() {
+			sb.waitForTuning()
+		}()
```

Comment:
similar to the SelectContextOrWait comment, we want to be able to stop this thread if possible

Reply:
just in one place should be enough. As long as we can get out of the for loop when we need to(i.e. stop the viam-server mid tuning)

Reply:
I think it would be better to add it to the for loop tbh